docker-style/action.yml

```diff
@@ -0,0 +1,133 @@
+name: >
+  Lint Dockerfile
+
+description: >
+  Evaluate the quality of your project Dockerfile(s) by using `hadolint
+  <https://github.com/hadolint/hadolint/>`_. This action is expected to
+  be used within a matrix job to lint Dockerfile(s) from multiple directories.
+  The action uses `hadolint-gh-action
+  <https://github.com/jbergstroem/hadolint-gh-action>`_ behind the scenes.
+  If you want to evaluate multiple Dockerfiles contained in various
+  directories of the provided directory, use the recursive option.
+  When linting a Dockerfile dedicated to Windows, one should use hadolint
+  shell pragma to avoid false positives from ShellCheck, see `hadolint shell pragma
+  <https://github.com/hadolint/hadolint/pull/708>`_.
+
+  .. note::
+
+      This action emphasizes the fact of having Dockerfile(s) contained inside the
+      'docker' directory in the root of the project.
+
+  .. warning::
+
+      This action only looks for docker files named Dockerfile.
+      A docker file like Dockerfile.linux will not be linted.
+
+inputs:
+
+  # Optional inputs
+
+  directory:
+    description: >
+      Directory from which to search for Dockerfile(s).
+    required: false
+    default: docker
+    type: string
+
+  recursive:
+    description: >
+      Search for Dockerfile(s) recursively.
+    required: false
+    default: false
+    type: bool
+
+  error-level:
+    description: >
+      Fail action based on hadolint output (-1: never, 0: error, 1: warning,
+      2: info)
+    required: false
+    default: 2
+    type: int
+
+runs:
+  using: "composite"
+  steps:
+
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Check that directory 'docker' exists at the root of the repository.
+          If it does not, then a warning is emitted.
+
+    - name: "Check if docker directory exists"
+      shell: bash
+      run: |
+        if [ -d "${{ github.workspace }}/docker" ]; then
+          echo "HAS_DOCKER_DIR=true" >> $GITHUB_ENV
+        else
+          echo "HAS_DOCKER_DIR=false" >> $GITHUB_ENV
+        fi
```

Comment:
This step is always performed... regardless the value of ``directory``. I think this is a bit misleading (aka anti-pattern)

I would do ``directory`` to be a string array, or None. If ``None``, then we default to docker if exists (we rely on the style action to disable the docker lint if we dont want to do docker linting).

Reply:
If you don't want to lint the docker directory then you can do it. Contrary to the previous approach, which would fail if the input directory was not `"docker"` or `".devcontainer"`, this new approach only triggers a warning saying 

```
No 'docker' directory found. Except in special cases, e.g. '.devcontainer', we emphasize that docker files are stored in the 'docker' directory at the root of the repository.
```

That's our way of reminding users that they should have a docker directory in which they store their docker files. If the message is not to your taste, what about

```
No 'docker' directory found. If possible, please follow the common approach of storing docker files in the 'docker' directory at the root of the repository.
```

Regarding the `directory` being a string array:
1. I think that the current approach already does what you propose since `docker-lint` can be used in a matrix job and defaults to `"docker"` if no input is provided
2. How would you write the action using a string array ? I can only think of using another script file which would add more disadvantages than advantages :(

Reply:
Well.. I don't want to discuss the directory structure, rather the need of having a matrix.

It seems that the `hadolint` action support multiple files: https://github.com/jbergstroem/hadolint-gh-action/blob/main/USAGE.md#pass-multiple-dockerfiles-to-process

If that is the case, then you can do something like:

```yml
  steps:
    - name: "Run PyAnsys docker style checks"
      uses: ansys/actions/docker-style@{{ version }}
      with:
        directory: |
            - docker
            - .devcontainer
            - my_other_docker_directory
```
The directory argument will be parsed as one string, replace the `\n *- `(regex) by one single space, and pass it to the `hadolint` action. This way, you avoid having one job per directory, because I presume that action does not render the worker useless, so it can be reused.

Reply:
I'll try it out and see how it does render in the end. I liked using matrix since it ends up with separated workflows but I hear your argument on resources / time consumption.
I'll test using multiple files at once and see how it works 👍

Reply:
Changing have been made to follow the requirement of handling multiple repo at once without using a github matrix. To use this feature one has to provide white space separated directories, e.g. "docker .devcontainer".

Reply:
thank you a lot @SMoraisAnsys :)